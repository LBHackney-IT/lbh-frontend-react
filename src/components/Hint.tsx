import classNames from "classnames";
import PropTypes from "prop-types";
import React from "react";

import "./Hint.scss";

/**
 * The proptypes for {@link Hint}.
 */
export interface HintProps {
  id?: string;
  className?: string;
  children: React.ReactNode;
}

/**
 * A component for providing hint text to users.
 */
export const Hint: React.FunctionComponent<HintProps> = (
  props: HintProps
): JSX.Element => {
  const { id, className, children } = props;

  return (
    <span id={id} className={classNames("govuk-hint", className)}>
      {children}
    </span>
  );
};

Hint.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};
