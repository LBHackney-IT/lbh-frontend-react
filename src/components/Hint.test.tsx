import React from "react";
import { create } from "react-test-renderer";

import { Hint } from "./Hint";

it("renders correctly with all props", () => {
  const component = create(
    <Hint id="test" className="class1 class2">
      Test
    </Hint>
  );

  expect(component).toMatchSnapshot();
});

it("renders correctly without optional props", () => {
  const component = create(<Hint>Test</Hint>);

  expect(component).toMatchSnapshot();
});
