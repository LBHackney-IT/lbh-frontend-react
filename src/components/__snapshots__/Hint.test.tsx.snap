// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<span
  className="govuk-hint class1 class2"
  id="test"
>
  Test
</span>
`;

exports[`renders correctly without optional props 1`] = `
<span
  className="govuk-hint"
>
  Test
</span>
`;
